src/palace/manager/api/circulation/base.py

```diff
@@ -98,8 +99,13 @@ def __init__(self, _db: Session, collection: Collection):
             )
 
     @property
-    def collection(self) -> Collection | None:
-        return Collection.by_id(self._db, id=self.collection_id)
+    def collection(self) -> Collection:
+        collection = Collection.by_id(self._db, id=self.collection_id)
+        if collection is None:
+            raise PalaceValueError(
+                f"Collection with id {self.collection_id} not found for {self.__class__.__name__}"
+            )
+        return collection
```

Comment:
I updated this type hint in the base class since all the implementations check if its none, then raise if it is. Instead of doing this in each implementation, I just moved this up to the base class, and removed the checks in each implementation.